### ISSUE TITLE
Design, implement and integrate a player stat struct into the existing implementation of the NPC struct.

### ISSUE THREAD

**Issue by avianurag** [uchicago-cs/chiventure#600]
The current NPC struct does not include a feature for player stats. This issue aims to plan out the exact structure used to store player stats (could be an enum like with player class), design the struct, and implement it into the NPC struct. The progress will be documented in this wiki:
https://github.com/uchicago-cs/chiventure/wiki/RPG-NPC:-Independent-Feature-Design-Document

**Comment by anshikab** [comment]
Update 1
Meeting on 05/16:

A meeting was held with a representative of the rpg_playerstats team that is working on a struct to store player stats. 
The code can be found here: https://github.com/uchicago-cs/chiventure/blob/rpg-stats/include/game-state/stats.h  
We discussed the integration of that struct into the NPC struct.  

- The stats_t struct can simply be included as a feature into the NPC struct.

- The NPC and stats_t structs both include a feature for health. This overlap may have to be addressed by removing the health feature from that NPC struct and just having it be a feature of the stats struct.

- The function npc_init that initializes an NPC will have to be modified to include this new feature. 

-Additional functionality that pertains to this feature (such as a function to obtain a player's stats) can be added as well.

**Comment by anshikab** [comment]
Code containing the relevant changes has been pushed to a new branch called rpg-npc/struct-features, which will either be merged with the dev branch once the npc module is merged, or with the npc branch before a cohesive pull request is submitted.

**Comment by anshikab** [comment]
I have updated the tests to account for the added feature.

**Comment by anshikab** [comment]
I have opened a pull request to merge with dev.

**Comment by anshikab** [comment]
This issue was linked to pull request #626  which has successfully been merged into dev. We were able to add stats functionality into the npc module by updating the npc struct, updating functions to create and initialize npc's, and updating test functions to accommodate rpg stats. This issue enabled us to integrate rpg related features into the npc module functioning to effectively link the two modules. Updates to the implementation of stats or npc's, or their integration will be incorporated in future issues. I am closing this issue.

**Comment by firatciftci** [comment]
Issue Score: ✔️++

Comments:
Great work!
